csharp/src/Google.Protobuf/MessageParser.cs

```diff
@@ -204,7 +234,7 @@ internal new T CreateTemplate()
         public new T ParseFrom(byte[] data)
         {
             T message = factory();
-            message.MergeFrom(data);
+            message.MergeFrom(data, DiscardUnknownFields);
```

Comment:
Do we have test coverage for these template?

Reply:
Not sure what you mean by "for these template" but the new test calls all the ParseFrom overloads, and the existing extension methods are tested by the existing tests.

Reply:
Sorry, I commented on wrong place. See comments above